src/main/java/org/adoptopenjdk/modulescanner/JdepsInspector.java

```diff
@@ -10,62 +13,88 @@
 import java.util.spi.ToolProvider;
 import java.util.stream.Collectors;
 
+/**
+ * This class is responsible for running jdeps on the JAR file it is inspecting
+ * and returning the result of that inspection
+ */
 public class JdepsInspector {
 
+    private static final Logger LOGGER = LogManager.getLogger("JdepsInspector");
+
     private Path jarFile;
 
     public JdepsInspector(Path jarFile) {
         this.jarFile = jarFile;
     }
 
+    /**
+     * Find the jdeps tool via the ToolsProvider, run it and then return a result
+     *
+     * @return The result of the jdeps inspection (could be a FAIL)
+     */
     public JdepsInspectResult inspect() {
         Optional<ToolProvider> jdepsTool = ToolProvider.findFirst("jdeps");
 
-        return jdepsTool.map(this::analyzeJdkInternals).orElse(JdepsInspectResult.FAIL_RESULT);
+        return jdepsTool.map(this::analyzeJdkInternals).orElse(JdepsInspectResult.FAIL);
     }
 
+    // Analyse the JDK internal API usage using jdeps
     private JdepsInspectResult analyzeJdkInternals(ToolProvider jdeps) {
+
         var outbytes = new ByteArrayOutputStream();
         var errbytes = new ByteArrayOutputStream();
 
         try {
-            int retVal = jdeps.run(new PrintStream(outbytes), new PrintStream(errbytes), "--jdk-internals", jarFile.toString());
+            int retVal = jdeps.run(new PrintStream(outbytes, true, Charset.defaultCharset()), new PrintStream(errbytes, true, Charset.defaultCharset()), "--jdk-internals", jarFile.toString());
 
-            String out = outbytes.toString(Charset.defaultCharset());
-            String err = errbytes.toString(Charset.defaultCharset());
+            String jdepsOutput = outbytes.toString(Charset.defaultCharset());
+            String errorOutput = errbytes.toString(Charset.defaultCharset());
 
-            if (retVal != 0 || (err != null && err.length() > 0)) {
-                return JdepsInspectResult.FAIL_RESULT;
+            if (retVal != 0 || (errorOutput != null && errorOutput.length() > 0)) {
+                return JdepsInspectResult.FAIL;
             }
 
-            return new JdepsInspectResult(false, getViolations(out));
+            return new JdepsInspectResult(false, getViolations(jdepsOutput));
         } catch (RuntimeException re) {
-            System.err.println("Could not process " + jarFile);
-            re.printStackTrace();
-            return JdepsInspectResult.FAIL_RESULT;
+            LOGGER.error("Could not process " + jarFile, re);
```

Comment:
This produces a large block of text. Expected but not nice: https://travis-ci.com/sandermak/modulescanner/jobs/132937400#L1379

Perhaps we can capture the logged error message in the test for `non-readable.jar`?

Reply:
Went with Trace for the stack trace